src/tracks/InstrumentTrack.cpp

```diff
@@ -860,15 +860,19 @@ void InstrumentTrack::saveTrackSpecificSettings( QDomDocument& doc, QDomElement
 
 	// Save the midi port info if we are not in song saving mode, e.g. in
 	// track cloning mode or if we are in song saving mode and the user
-	// has chosen to discard the MIDI connections.
+	// has chosen not to discard the MIDI connections.
 	if (!Engine::getSong()->isSavingProject() ||
 	    !Engine::getSong()->getSaveOptions().discardMIDIConnections.value())
 	{
 		// Don't save auto assigned midi device connection
 		bool hasAuto = m_hasAutoMidiDev;
 		autoAssignMidiDevice(false);
 
-		m_midiPort.saveState( doc, thisElement );
+		// Only save the MIDI port information if we are not saving a preset.
+		if (!isPresetMode())
+		{
+			m_midiPort.saveState( doc, thisElement );
```

Comment:
style, no space with the brackets

Reply:
Fixed with commit https://github.com/LMMS/lmms/pull/7445/commits/c822c10a68a512392f12233987634b515b74b1a4.